santoku/utils/url_handler.py

```diff
@@ -440,16 +611,43 @@ def explode_domain(self) -> List[str]:
         return list(reversed(exploded_domains))
 
     @classmethod
-    def explode_path(cls, path: str, max_depth: int = 1) -> List[str]:
-        """Given a path, returns a list with exploded paths up to the specified `max_depth`.
+    def explode_path(cls, path: Optional[str], max_depth: int = 1) -> List[str]:
+        """
+        Given a path, returns a list with exploded paths up to the specified `max_depth`.
+
+        We consider the path depth as the number of components of a `path` split on '/'.
+
+        Parameters
+        ----------
+        path : Optional[str]
+            Path to be exploded.
+
+        max_depth : int, Optional
+            Depth of paths to be considered in the path exploitation. By default, 1 level of `path`
+            is exploded.
+
+        Returns
+        -------
+        List[str]
+            The list of exploded paths up to `max_depth`, or an empty list if `path` is empty. If
+            `max_depth` is greater than the number of components of the path, explodes the path with
+            all components. If `max_depth` is smaller than 1, an empty list will be returned.
+
+        See also
+        --------
+        clean_component
+            Called to clean the path.
 
         Examples
         --------
         "here/there", 1 -> ["here"]
         "here/there", 2 -> ["here", "here/there"]
+
         """
-        if path is None or path in ["", "/"]:
+        cleaned_path = cls.clean_component(component=path, lowercase=False)
+
+        if not cleaned_path:
             return []
 
-        path_parts = path.strip("/").split("/")
+        path_parts = cleaned_path.split("/")
         return ["/".join(path_parts[: i + 1]) for i in range(min(max_depth, len(path_parts)))]
```

Comment:
🤯 
Don't use list comprehension in this case. It's hard to tell what is going on.